CenterForOpenScience/osf.io
# [ENG-2579][ENG-2580] Add IA integration

## Purpose

This is intended to make sure our IA buckets have their metadata synced with their corresponding registrations and that files are archived on IA when registered.

## Changes

- add post_save signals to update bucket
- add signal when registrations are made public, either immediately or after embargo
- add tests and mocks


## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify change registration metadata and see that the IA item changes.
- Verify files are uploaded to internet archive on registration

## Documentation



new behavior, no docs yet.

## Side Effects

None that I know of.

## Ticket

Sync Metadata
https://openscience.atlassian.net/browse/ENG-2580

Archive on Registration
https://openscience.atlassian.net/browse/ENG-2579
